### PR TITLE
Slim sidebar UI and accessibility fixes – take two

### LINKED ISSUE #7742: Slim sidebar always showing scrollbars
On Wagtail 2.15.1, with Firefox 93, scrollbars are always showing on the slim sidebar, causing the logo to be clipped:

image

### PR DESCRIPTION
**Pull request by thibaudcolas**
Follow-up to #7844. Again this fixes four separate issues reviewable in the separate commits.

- Fixes #7742 (reproduce-able in Windows and Linux, not macOS)
- branding_logo support for slim sidebar
- Fix focus outline issues I picked up while doing further high-contrast mode testing
- Misc small high-contrast mode improvements

Here is the branding_logo definition I used to test this, to add in `templates/wagtailadmin/base.html` in bakerydemo:

```html
{% extends "wagtailadmin/base.html" %}

{% block branding_logo %}
    <img src="https://media.wagtail.io/images/default-sharing-image.width-1024.png" alt="Custom Project" width="80" />
{% endblock %}
```

And the menu item customisations for reference (no changes relevant to this in this PR):

```py
from wagtail.admin.menu import MenuItem
from wagtail.core import hooks


@hooks.register('register_settings_menu_item')
def register_locales_menu_item():
    return MenuItem("Test settings", "/test/", icon_name='site', classnames="1234-settings", order=603, attrs={'data-test': '1234-settings'})

@hooks.register('register_admin_menu_item')
def register_locales_menu_item():
    return MenuItem("Test admin", "/test/", icon_name='site', classnames="1234-admin", order=603, attrs={'data-test': '1234-admin'})

@hooks.register('register_reports_menu_item')
def register_locales_menu_item():
    return MenuItem("Test reports", "/test/", icon_name='site', classnames="1234-reports", order=603, attrs={'data-test': '1234-reports'})


@hooks.register('construct_main_menu')
def construct_main_menu(request, menu_items):
    menu_items.append(
        MenuItem('construct_main_menu', '/kittens/', classnames='construct_main_menu-1234', order=1000, attrs={'data-test': '1234-main'})
    )

@hooks.register('construct_settings_menu')
def construct_main_menu(request, menu_items):
    menu_items.append(
        MenuItem('construct_settings_menu', '/kittens/', classnames='construct_settings_menu-1234', order=1000, attrs={'data-test': '1234-settings'})
    )


@hooks.register('construct_reports_menu')
def construct_main_menu(request, menu_items):
    menu_items.append(
        MenuItem('construct_reports_menu', '/kittens/', classnames='construct_reports_menu-1234', order=1000, attrs={'data-test': '1234-reports'})
    )
```

* [x] Do the tests still pass? (https://docs.wagtail.io/en/latest/contributing/developing.html#testing)
* [x] Does the code comply with the style guide? (Run `make lint` from the Wagtail root)
* ~[ ] For Python changes: Have you added tests to cover the new/fixed behaviour?~ N/A
* [x] For front-end changes: Did you test on all of [Wagtail’s supported environments](https://docs.wagtail.io/en/latest/contributing/developing.html#browser-and-device-support)?
    * Firefox 91 ESR Windows 11, MS Edge 94 Windows 10, macOS Safari latest, macOS Chrome latest, iOS Safari 15, Android 11 Chrome
    * Windows high-contrast mode, macOS VoiceOver, NVDA
* ~[ ] For new features: Has the documentation been updated accordingly?~ N/A

### CODE REVIEW

File: wagtail/admin/templates/wagtailadmin/base.html [hunk 1]
```diff
@@ -3,21 +3,20 @@
 
 {% block furniture %}
     {% slim_sidebar_enabled as slim_sidebar_enabled %}
+    <template data-wagtail-sidebar-branding-logo>{% block branding_logo %}{% endblock %}</template>
```
Comment (thibaudcolas) [inline wagtail/admin/templates/wagtailadmin/base.html:6]: This and below – refactoring was needed so we could retain the single `branding_logo` block. The backwards-compatible implementation with the inline script tag for the legacy sidebar is pretty poor, but hopefully good enough for a single release.

File: wagtail/admin/templates/wagtailadmin/base.html [hunk 2]
```diff
@@ -26,6 +25,16 @@
         </div>
         <div class="explorer__wrapper" data-explorer-menu></div>
     </aside>
+    {# Backwards-compatibility for branding_logo customisations in legacy sidebar. #}
+    {# TODO Remove in Wagtail 2.17 #}
```
Comment (kaedroho) [inline wagtail/admin/templates/wagtailadmin/base.html:29]: I'd add ``RemovedInWagtail217Warning`` to your comment here so Matthew can grep for this later

Comment (gasman) [reply wagtail/admin/templates/wagtailadmin/base.html:29]: Cheers! (Won't the removal happen in 2.18 now, though?)

**Review by kaedroho — APPROVED** [review]
This looks good to me, thanks! (just a minor suggestion to make finding the bit of code to remove later)

File: wagtail/admin/templates/wagtailadmin/base.html (hunk 2 above)
Comment (thibaudcolas) [reply wagtail/admin/templates/wagtailadmin/base.html:29]: @gasman I was expecting the following sequence:

- v2.15 (current) – slim sidebar behind feature flag (opt-in)
- v2.16 (upcoming) – slim sidebar on by default, can opt-out to legacy sidebar
- v2.17 (upcoming + 1) – legacy sidebar removed

Does that look right?

Comment (thibaudcolas) [reply wagtail/admin/templates/wagtailadmin/base.html:29]:
```suggestion
    {# TODO RemovedInWagtail217Warning Remove in Wagtail 2.17 #}
```

**Pull request merged**